datahub-web-react/src/app/entity/dataset/profile/schema/SchemaDescriptionField.tsx

```diff
@@ -52,23 +77,26 @@ export default function DescriptionField({ description, updatedDescription, onUp
                 e.stopPropagation();
             }}
         >
-            <DescriptionText
-                source={updatedDescription || description}
-                editable
-                onEditClicked={() => setShowAddModal(true)}
-            />
-            {updatedDescription && <EditedLabel>(edited)</EditedLabel>}
+            <DescriptionText source={currentDesc} />
+            <EditIcon twoToneColor="#52c41a" onClick={() => setShowAddModal(true)} />
+            {(!!updatedDescription || updatedDescription === '') && <EditedLabel>(edited)</EditedLabel>}
             {showAddModal && (
                 <div>
                     <UpdateDescriptionModal
-                        title="Update description"
-                        description={updatedDescription || description}
+                        title={currentDesc ? 'Update description' : 'Add description'}
+                        description={currentDesc}
                         original={description}
                         onClose={onCloseModal}
                         onSubmit={onUpdateModal}
+                        isAddDesc={currentDesc === null}
                     />
                 </div>
             )}
+            {currentDesc === null && (
```

Comment:
```suggestion
            { (!currentDesc || currentDesc === '') && (
```

this should cover a few more cases